os/mm/mm_heap/mm_getheap.c

```diff
@@ -51,9 +51,93 @@ extern struct mm_heap_s g_mmheap[CONFIG_MM_NHEAPS];
  * Pre-processor Definitions
  ****************************************************************************/
 
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+typedef struct app_heap_s {
+	struct mm_heap_s *heap;
+	struct app_heap_s *next;
+	struct app_heap_s *prev;
+} app_heap_s;
+
+static app_heap_s *app_heap_list;
+extern struct mm_heap_s g_pheap;
+#endif
+
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+void mm_add_app_heap_list(struct mm_heap_s *heap)
+{
+	app_heap_s *node = (app_heap_s *)kmm_malloc(sizeof(app_heap_s));
+	if (!node) {
+		mdbg("Error allocating heap node\n");
+		return;
+	}
+
+	node->heap = heap;
+
+	/* Add the new heap node to the head of the list*/
+	node->next = app_heap_list;
```

Comment:
i think app_heap_list prev should be assigned to current node?

Reply:
Yes. Will fix.